src/app/rosetta/lib/block.ml

```diff
@@ -0,0 +1,579 @@
+      ; fee_payer= `Pk "Alice"
+      ; source= `Pk "Alice"
+      ; token= Unsigned.UInt64.of_int 1
+      ; fee= Unsigned.UInt64.of_int 2_000_000_000
+      ; receiver= `Pk "Bob"
+      ; fee_token= Unsigned.UInt64.of_int 1
+      ; nonce= Unsigned.UInt32.of_int 3
+      ; amount= None
+      ; failure_status=
+          Some
+            (`Applied
+              (Account_creation_fees_paid.By_fee_payer
+                 (Unsigned.UInt64.of_int 3_000)))
+      ; hash= "TXN_3b_HASH" }
+    ; { kind= `Create_account
+      ; fee_payer= `Pk "Alice"
+      ; source= `Pk "Alice"
+      ; token= Unsigned.UInt64.of_int 1
+      ; fee= Unsigned.UInt64.of_int 2_000_000_000
+      ; receiver= `Pk "Bob"
+      ; fee_token= Unsigned.UInt64.of_int 1
+      ; nonce= Unsigned.UInt32.of_int 3
+      ; amount= None
+      ; failure_status= Some (`Applied Account_creation_fees_paid.By_no_one)
+      ; hash= "TXN_4_HASH" }
+    ; { kind= `Mint_tokens
+      ; fee_payer= `Pk "Alice"
+      ; source= `Pk "Alice"
+      ; token= Unsigned.UInt64.of_int 10
+      ; fee= Unsigned.UInt64.of_int 2_000_000_000
+      ; receiver= `Pk "Bob"
+      ; fee_token= Unsigned.UInt64.of_int 1
+      ; nonce= Unsigned.UInt32.of_int 3
+      ; amount= Some (Unsigned.UInt64.of_int 30_000)
+      ; failure_status= Some (`Applied Account_creation_fees_paid.By_no_one)
+      ; hash= "TXN_5_HASH" } ]
+end
+
+module Internal_command_info = struct
+  module Kind = struct
+    type t = [`Coinbase | `Fee_transfer] [@@deriving eq]
+  end
+
+  type t =
+    { kind: Kind.t
+    ; receiver: [`Pk of string]
+    ; fee: Unsigned.UInt64.t
+    ; token: Unsigned.UInt64.t
+    ; hash: string }
+
+  let to_operations ~user_commands ~coinbase ~block_creator_pk (t : t) :
+      Operation.t list =
+    (* We choose to represent the dec-side of fee transfers from txns from the
+     * canonical user command that created them so we are able consistently
+     * produce more balance changing operations in the mempool or a block.
+     * *)
+    let plan : 'a Op.t list =
+      match t.kind with
+      | `Coinbase ->
+          [{Op.label= `Coinbase_inc; related_to= None}]
+      | `Fee_transfer ->
+          (* Detect if a fee transfer comes from a snark worker payment by
+         * failing to find the user command that caused it *)
+          if
+            List.find user_commands ~f:(fun (info : User_command_info.t) ->
+                Unsigned.UInt64.equal info.fee t.fee
+                && Unsigned.UInt64.equal info.token t.token
+                && [%eq: [`Pk of string]] info.receiver t.receiver )
+            |> Option.is_none
+          then
+            (* In this case we make a "synthetic" fee subtraction from block
+           * creator even though this is implicit in our blocks. We need this
+           * so the balance always sums to zero. *)
+            [ {Op.label= `Fee_payer_dec; related_to= None}
+            ; {Op.label= `Fee_receiver_inc; related_to= Some `Fee_payer_dec} ]
```

Comment:
As discussed, this conditional should be changed. Copying my comment from the other PR for reference:
How are Internal_command_info.ts generated? If it maps one-to-one with the fee transfers and coinbase transactions added to the scan state (or archive database) then this wouldn't work. User command fees are always aggregated and sent to the coinbase receiver as a single fee transfer(minus the prover fees). The operation Fee_payer_dec should be added for each user_command in that case and Fee_receiver_inc related to Fee_payer_dec would be if the receiver is the coinbase receiver

Reply:
I think I'm confused where to put the negative ops and the positive ones:

Is the coinbase reward command distinct from the fee transfer giving the `user-command-fees - prover fees` to the coinbase receiver? Or is it a separate fee transfer?

Reply:
`Fee_receiver_inc` ops would come from the fee transfers. `Fee_payer_dec` ops would come from user commands. So for example: 
Say we have three user commands with fee 1, 2, 3 respectively, a coinbase transaction, and fee transfers (prover1, fee=1), (coinbase_rec, fee=5) in a block
The fee-related ops would then be:
1. Three `Fee_payer_dec`s with amount -1, -2, and -3 respectively (These would be generated from the user commands)
2. One coinbase op
3. Two `Fee_receiver_inc`s with amount 1 (receiver: prover1) and 5 (receiver: coinbase_receiver) respectively